server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -240,6 +364,35 @@ private void writeDefaultValue(StreamOutput out, Object defaultValue) throws IOE
         }
     }
 
+    private Object readParser(StreamInput in, Object parser) throws IOException {
+        switch (type) {
+            case Boolean:
+                return new BooleanParser(in);
+            case Integer:
+                return new IntegerParser(in);
+            case Long:
+                return new LongParser(in);
+            case Float:
+                return new FloatParser(in);
+            case Double:
+                return new DoubleParser(in);
+            case TimeValue:
+                if (in.readBoolean()) {
+                    return new MinMaxTimeValueParser(in);
+                } else {
```

Comment:
Match it with `writeParser` above and change `else` to `else if` for strict checking